src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,28 +111,38 @@ data CCurrency
     deriving (Show, Read, Generic)
 
 -- | Client hash
-newtype CHash = CHash Text deriving (Show, Eq, Generic, Buildable)
+newtype CHash = CHash Text
+    deriving (Show, Eq, Generic, Buildable)
 
 instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash
```

Comment:
Oh, I got your point. I'm not for or against it, but even if we'd want to allow only two types there, how would you do that? (I didn't manage find needed example beyond the link 😿 )